contracts/staking/PrtStakingPool.sol

```diff
@@ -0,0 +1,258 @@
+ */
+contract PrtStakingPool is Ownable, ERC20Snapshot, ReentrancyGuard {
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event FeeSplitExtensionChanged(address _newFeeSplitExtension);
+
+    /* ============ State Variables ============ */
+
+    /// @notice SetToken to be distributed to stakers
+    ISetToken public immutable setToken;
+
+    /// @notice PRT token to be staked
+    IPrt public immutable prt;
+
+    /// @notice PRT Fee split extension which accrues fees and distributes setToken
+    address public feeSplitExtension;
+
+    /// @notice Snapshot id of the last claim for each staker
+    mapping(address => uint256) public lastSnapshotId;
+
+    /// @notice Amount of setToken accrued and distributed with each snapshot
+    uint256[] public accrueSnapshots;
+
+    /* ============ Modifiers ============ */
+
+    /**
+     * @dev Modifier to restrict snapshot calls to only the fee split extension.
+     */
+    modifier onlyFeeSplitExtension() {
+        require(msg.sender == feeSplitExtension, "Must be FeeSplitExtension");
+        _;
+    }
+
+    /* ========== Constructor ========== */
+
+    /**
+     * @notice Constructor to initialize the PRT Staking Pool.
+     * @param _name Name of the staked PRT token
+     * @param _symbol Symbol of the staked PRT token
+     * @param _prt Instance of the PRT token contract
+     * @param _feeSplitExtension Address of the PrtFeeSplitExtension contract
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        IPrt _prt,
+        address _feeSplitExtension
+    )
+        public
+        ERC20(_name, _symbol)
+    {
+        prt = _prt;
+        setToken = ISetToken(_prt.setToken());
+        feeSplitExtension = _feeSplitExtension;
+    }
+
+    /* ========== External Functions ========== */
+
+    /**
+     * @notice Stake `amount` of PRT tokens from `msg.sender` and mint staked PRT tokens.
+     * @param _amount The amount of PRT tokens to stake
+     */
+    function stake(uint256 _amount) external nonReentrant {
+        require(_amount > 0, "Cannot stake 0");
+        prt.transferFrom(msg.sender, address(this), _amount);
+        super._mint(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Unstake `amount` of PRT tokens by `msg.sender`.
+     * @param _amount The amount of PRT tokens to unstake
+     */
+    function unstake(uint256 _amount) public nonReentrant {
+        require(_amount > 0, "Cannot unstake 0");
+        super._burn(msg.sender, _amount);
+        prt.transfer(msg.sender, _amount);
+    }
+
+    /**
+     * @notice ONLY FEE SPLIT EXTENSION: Accrue SetTokens and update snapshot.
+     * @param _amount The amount of SetTokens to accrue
+     */
+    function accrue(uint256 _amount) external nonReentrant onlyFeeSplitExtension {
+        setToken.transferFrom(msg.sender, address(this), _amount);
+        accrueSnapshots.push(_amount);
+        super._snapshot();
```

Comment:
this will emit a `PrtStakingPool.Snapshot` event with the snapshotId, see added tests here [`d30288a`](https://github.com/IndexCoop/index-coop-smart-contracts/pull/176/commits/d30288ae3ceafd5ba65f6d738cc8d66234e3044b)

it will also emit a `PrtFeeSplitExtension.PrtFeesDistributed` event and `SetToken.Transfer` event from the `PrtFeeSplitExtension` to the `PrtStakingPool`